doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,495 @@
+
+* DPU local stateDB Schema for DPU_HEALTH
+    ```
+    Table: “DPU_HEALTH”
+
+    SCHEMA
+    key:  dpu_health
+    
+    HMSET dpu_health 
+        "value": { 
+            "count": "1",  # number of occurrence of event 
+            "description": "Single bit error Correction", # Event
+            "name": "ms.ms.int_prp2_read", 
+            "severity": "LEVEL_INFO", # DEBUG, INFO, WARNING, ERROR
+            "timestamp": "20230618 14:56:15"
+        } 
+    ```
+* ModuleBase class new APIs
+
+    | API | current_usage | switch cpu | dpu | SONiC enhancements | PD plugin change| HW mgmt change | comments |
+    | --- | --- | --- | --- | --- | --- | --- | --- |
+    | get_state_info(self, index) | Retrieves the dpu state object having the detailed dpu state progression | yes | no | yes | yes | yes | Fetched from ChassisStateDB |
+    | get_health_info(self) | Retrieves the dpu health object | no | yes | yes | yes | yes | Fetched from DPU |
+
+### 3.2. Thermal management
+* Platform  initializes all sensors
+* Thermalctld fetch CPU temperature, DPU temperature, fan speed, monitor and update the DB
+* Thermal manager reads all thermal sensor data, run thermal policy and take policy action Ex. Set fan speed, set alarm, set syslog, set LEDs 
+* Platform collects fan related data such as presence, failure and then applies fan algorithm to set the new fan speed
+* The north bound CLI/Utils/App use DB data to ”show environment”, ”show platform temp” show platform fan”
+
+Thermal management sequence diagram
+<p align="center"><img src="./images/thermal-mgmt-seq.svg"></p>
+
+### 3.2.1 Platform device data collection 
+* thermalctld, led and PSUd post device data to DB periodically
+* during the boot up of the daemons, it will collect the constant data like serial number, manufacture name, etc.
+* For the variable ones (temperature, voltage, fan speed ....) need to be collected periodically. 
+
+### 3.3.   Midplane Interface
+A typical modular chassis includes a midplane-interface to interconnect the Supervisor & line-cards. The same design has been extended in case of a SmartSwitch. The mnic ethernet interface over PCIe which is the midplane-interface, interconnect the Switch Host and the DPUs.
+
+* When DPU card or the Supervisor boots and as part of its initialization, midplane interface gets initialized.
+* Two solutions were proposed for this purpose.  
+    * Solution 1. uses a dhcp server - client model between the switch host and the dpus to allocate IP address. [link](https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/ip-address-assigment/smart-switch-ip-address-assignment.md)
+    * Solution 2. static ip allocation based on DPU ID. [link](https://github.com/rameshraghupathy/SONiC/blob/origin/ss_pmon_hld/doc/smart-switch/pmon/static-ip-assignment.md)
```

Comment:
@rameshraghupathy can we remove Solution 2.

Reply:
Done